src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1077,7 +1110,15 @@ internal unsafe int IndexOf(string source, string value, int startIndex, int cou
             }
             else
             {
-                return IndexOfCore(source, value, startIndex, count, options, matchLengthPtr);
+                if (fromBeginning)
+                {
+                    // Call the string-based overload, as it special-cases IsFastSort as a perf optimization.
```

Comment:
Are the changes in this file needed for the Utf8String? It looks unrelated. It would be best to submit it as separate change. Also, I think we should be rather working get rid of the IsFastSort, not coding around it.

Reply:
Yes, it's needed as the core "string search" routine, in particular for anything other than Ordinal. See https://github.com/GrabYourPitchforks/coreclr/blob/bfbeb452b91825843ccd75eb37cff749996fd4af/src/System.Private.CoreLib/src/System/Text/Utf8Span.Searching.cs#L275 for the particular call site that uses this.

I'm fine with getting rid of `IsFastSort`, but it seems like that's best left as a separate PR? I've opened https://github.com/dotnet/coreclr/issues/26758 to track.